lib/node_modules/@stdlib/fs/resolve-parent-paths/lib/async.js

```diff
@@ -0,0 +1,425 @@
+				if ( child === dir ) {
+					return done( null, out );
+				}
+				// Resolve paths at next directory level:
+				for ( j = 0; j < paths.length; j++ ) {
+					spath = resolve( dir, paths[ j ] );
+					exists( spath, getCallback( spath ) );
+				}
+			}
+		}
+	}
+}
+
+/**
+* Asynchronously resolves all paths from a set of paths at a directory level by walking parent directories.
+*
+* @private
+* @param {Array<string>} paths - paths to resolve from
+* @param {string} dir - base directory
+* @param {Callback} done - callback to invoke after resolving paths
+* @returns {void}
+*/
+function all( paths, dir, done ) {
+	var count;
+	var child;
+	var spath;
+	var FLG = 0; // flag to track if we are done traversing a directory level
+	var out;
+	var i;
+
+	count = 0;
+	out = filled( null, paths.length );
+
+	// Start at a base directory and continue moving up through each parent directory...
+	for ( i = 0; i < paths.length; i++ ) {
+		spath = resolve( dir, paths[ i ] );
+		exists( spath, getCallback( i, spath ) );
+	}
+
+	/**
+	* Returns a callback to be invoked upon checking for path existence.
+	*
+	* @private
+	* @param {NonNegativeInteger} idx - index
+	* @param {string} spath - resolved path
+	* @returns {Callback} callback
+	*/
+	function getCallback( idx, spath ) {
+		return onExists;
+
+		/**
+		* Callback invoked after checking for path existence.
+		*
+		* @private
+		* @param {(Error|null)} error - error object
+		* @param {boolean} bool - boolean indicating if a path exists
+		* @returns {void}
+		*/
+		function onExists( error, bool ) { // eslint-disable-line node/handle-callback-err
+			var j;
+			if ( bool ) {
+				out[ idx ] = spath;
+				count += 1;
+			}
+			// If we have traversed all paths at the current directory level, resolve parent directory...
+			FLG += 1;
+			if ( FLG === paths.length ) {
+				// Check if we have resolved any path...
+				if ( count === paths.length ) {
+					return done( null, out );
+				}
+				// Resolve a parent directory:
+				child = dir;
+				dir = resolve( dir, '..' );
+
+				// Reset flag and buffers:
+				FLG = 0;
+				out = [];
+				count = 0;
+
+				// If we have already reached root, we cannot resolve any higher directories...
+				if ( child === dir ) {
+					return done( null, out );
+				}
+				// Resolve paths at next directory level:
+				for ( j = 0; j < paths.length; j++ ) {
+					spath = resolve( dir, paths[ j ] );
+					exists( spath, getCallback( j, spath ) );
+				}
+			}
+		}
+	}
+}
+
+/**
+* Asynchronously resolves each path from a set of paths by walking parent directories.
+*
+* @private
+* @param {Array<string>} paths - paths to resolve from
+* @param {string} dir - base directory
+* @param {Callback} done - callback to invoke after resolving paths
+* @returns {void}
+*/
+function each( paths, dir, done ) {
+	var count;
+	var child;
+	var spath;
+	var FLG = false; // flag to check if we are done resolving all paths
+	var out;
+	var i;
```

Comment:
In general, throughout the codebase, unless all variable declarations are assigned while still retaining line length order, no declarations should be assigned. Hence, declaration and initialization are treated as separate steps.
```suggestion
	var count;
	var child;
	var spath;
	var FLG;
	var out;
	var i;
	
	FLG = false; // flag to check if we are done resolving all paths
```